content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,161 @@
+---
+title: Saving
+---
+
+GitHub's save patterns help developers to store and update their configurations for different features throughout GitHub's various interfaces. These changes should be represented in the UI accurately, quickly, and obviously. Their behavior should be easy to reason about and inspire confidence and trust.
+
+All configuration flows should be punctuated with CTAs that feature a logical term such as "Create", "Save", "Delete", or "Update" (e.g. Creating a new repository, adding an SSH key to your profile, saving security preferences, editing a repository's description). 
+
+Individual settings that are not distinct flows should minimize the required UI by saving configuration changes as they are made (e.g. changing your primary email address, adding an assignee to an issue, unsubscribing from a notification type, following a user, watching a repository)
+
+There are two primary patterns that should be used for saving user input on GitHub:
+  - Save and route
+    - This pattern is primarily used for creation and deletion of new entities such as repositories, organizations, or user accounts. At the end of the flow, the user should be routed to the entity they just created. In some cases such as forking or using a template, we may need to hold the user at a waiting page while the entity is generated. In these cases, we should make the user feel confident that something is happening and that their configuration details will not be lost. In some cases, it may make sense to be explicit that the interaction has occurred by using either a [Toast]() or a [Flash]() alert.
+    - [Example code]
+  - Optimistic update
+    - For interactions where the entity already exists and we merely need to update a field on that object, we should fire off the request to the server, immediately display the desired changes as if they've already happened, and upon response from the server, either cache the result of a successful change to save loading time later or present a clear, concise error message if an error has occurred. If the change will not be obviously visible in the UI the user is currently viewing, we should fire off a [Toast]() to let the user know that the change occurred successfully.
```

Comment:
Is there a better term for "fire off"?

Reply:
what do you think of "trigger"? maybe "initiate"? I try to follow the rules of good prose and not repeat the same word *too* often, so if you've got suggestions, I'm open!

Reply:
I like initiate.